android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/SelectLanguageFragment.java

```diff
@@ -190,6 +190,41 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
     listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
 
+      /**
+       * Utility to modify keyboardList. Based on BCP47.toggleLanguage
+       * If keyboard's languageID exists in the list, remove the keyboard.
+       * Otherwise, add the keyboard to the list.
+       * @param keyboardList
+       * @param k
+       */
+      private void toggleLanguage(ArrayList<Keyboard> keyboardList, Keyboard k) {
+        if (keyboardList != null) {
```

Comment:
See my other comment on `toggleLanguage` on validating input parameters. Furthermore, you are only testing `keyboardList` and not `k` which is certainly insufficient :grin: